src/vectara/client.py

```diff
@@ -1,12 +1,96 @@
-from .base_client import BaseVectara, AsyncBaseVectara
+from . import SearchCorporaParameters, GenerationParameters, ChatParameters, ChatStreamedResponse
+from .base_client import BaseVectara, AsyncBaseVectara, OMIT
 from vectara.managers.corpus import CorpusManager
 from vectara.managers.upload import UploadManager
 from vectara.managers.document import DocumentManager
 from vectara.utils import LabHelper
 
-from typing import Union, Optional, Callable
+from typing import Union, Iterator
 import logging
 
+from typing import Optional
+
+from .core import RequestOptions
+
+
+class ChatSession:
+    def __init__(
+            self,
+            client,
+            search: SearchCorporaParameters,
+            chat_id: Optional[str] = None,
+            request_timeout: Optional[int] = None,
+            request_timeout_millis: Optional[int] = None,
+            generation: Optional[GenerationParameters] = OMIT,
+            chat_config: Optional[ChatParameters] = OMIT,
+            request_options: Optional[RequestOptions] = None,
+    ):
+        self.client = client
+        self.chat_id = chat_id
+        self.search = search
+        self.generation = generation
+        self.chat_config = chat_config
```

Comment:
Do we need chat_config? It's got only a single parameter (whether to store chat or not from the API), so I think it's redundant. When you create a ChatSession - you always want that to be true - wdyt?

Reply:
I think lets keep it because API does have the option to store it or not. Some users might not want to store the chat.